logic/broadcaster.js

```diff
@@ -242,21 +247,21 @@ __private.filterQueue = function(cb) {
 };
 
 /**
- * Checks if transaction is in pool or confirm it.
+ * Checks if transaction of a given id is in pool or confirmed in database "trs" table.
  *
  * @private
- * @param {transaction} transaction
+ * @param {string} transactionId
  * @param {function} cb
  * @returns {SetImmediate} null, boolean
  * @todo Add description for the params
  */
-__private.filterTransaction = function(transaction, cb) {
-	if (transaction !== undefined) {
-		if (modules.transactions.transactionInPool(transaction.id)) {
+__private.filterTransaction = function(transactionId, cb) {
+	if (transactionId !== undefined) {
+		if (modules.transactions.transactionInPool(transactionId)) {
 			return setImmediate(cb, null, true);
 		}
 		return library.logic.transaction.checkConfirmed(
-			transaction,
+			{ id: transactionId },
```

Comment:
So `checkConfirmed` is expecting a transaction object, now from here we are sending an object with just `id` property. This creates inconsistency and can be an issue in the future. Consider changing this or modify documentation of `checkConfirmed` to explicitly say it's not a transaction object.